### ISSUE TITLE
    assert x.dim() == 2 and x.size(1) == self.in_features KeyboardInterrupt

### ISSUE THREAD

**Issue by ckq11903** [akaashdash/kansformers#1]
提示这个错误呢，是维度不匹配吗

**Comment by vmallela0** [comment]
no, this is being caused by keyboard interruption it seems. are you doing control c when running?

**Comment by vmallela0** [comment]
yes, it's checking the dimension of the tensor, seeing if it's 2 (and then the second value's dimensions to see if we have the correct number of inputs)

**Comment by ckq11903** [comment]
> yes, it's checking the dimension of the tensor, seeing if it's 2 (and then the second value's dimensions to see if we have the correct number of inputs)是的，它正在检查张量的维度，看看它是否是 2（然后是第二个值的维度，看看我们是否有正确的输入数量）

So how do I need to modify it?,In this code is entered in three-dimensional

**Comment by akaashdash** [comment]
> > yes, it's checking the dimension of the tensor, seeing if it's 2 (and then the second value's dimensions to see if we have the correct number of inputs)是的，它正在检查张量的维度，看看它是否是 2（然后是第二个值的维度，看看我们是否有正确的输入数量）
> 
> So how do I need to modify it?,In this code is entered in three-dimensional

We are using efficient-kan with some modification to allow shape compatibility. Changes can be seen here: https://github.com/Blealtan/efficient-kan/pull/6
